qcodes/instrument_drivers/Andor/DU401.py

```diff
@@ -0,0 +1,424 @@
+
+    def set_number_accumulations(self, number):
+        c_number = ctypes.c_int(number)
+        code = self.dll.SetNumberAccumulations(c_number)
+        self.error_check(code, 'SetNumberAccumulations')
+
+    def set_read_mode(self, mode):
+        code = self.dll.SetReadMode(mode)
+        self.error_check(code, 'SetReadMode')
+
+    def set_shutter(self, typ, mode, closing_time, opening_time):
+        c_typ = ctypes.c_int(typ)
+        c_mode = ctypes.c_int(mode)
+        c_closing_time = ctypes.c_int(closing_time)
+        c_opening_time = ctypes.c_int(opening_time)
+        code = self.dll.SetShutter(c_typ, c_mode, c_closing_time, c_opening_time)
+        self.error_check(code, 'SetShutter')
+
+    def set_temperature(self, temperature):
+        c_temperature = ctypes.c_int(temperature)
+        code = self.dll.SetTemperature(c_temperature)
+        self.error_check(code, 'SetTemperature')
+
+    def set_trigger_mode(self, mode):
+        c_mode = ctypes.c_int(mode)
+        code = self.dll.SetTriggerMode(c_mode)
+        self.error_check(code, 'SetTriggerMode')
+
+    def shut_down(self):
+        code = self.dll.ShutDown()
+        self.error_check(code, 'ShutDown')
+
+    def start_acquisition(self):
+        code = self.dll.StartAcquisition()
+        self.error_check(code, 'StartAcquisition')
+
+    def wait_for_acquisition(self):
+        code = self.dll.WaitForAcquisition()
+        self.error_check(code, 'WaitForAcquisition')
+
+
+class Spectrum(Parameter):
+    """
+    Parameter class for a spectrum taken with an Andor CCD.
+
+    The spectrum is saved in a list with the length being set by the number of pixels on the CCD.
+
+    Args:
+        name (str): Parameter name.
+
+    """
+
+    def __init__(self, name, *args, **kwargs):
+
+        self.ccd = kwargs['instrument']
+        super().__init__(name, *args, **kwargs)
+
+    def get_raw(self):
+
+        # get acquisition mode
+        acquisition_mode = self.ccd.acquisition_mode.get()
+
+        # start acquisition
+        self.ccd.atmcd64d.start_acquisition()
+
+        # wait for single acquisition
+        if acquisition_mode == 'single scan':
+            self.ccd.atmcd64d.wait_for_acquisition()
+
+        # wait for accumulate acquisition
+        elif acquisition_mode == 'accumulate':
+            number_accumulations = self.ccd.number_accumulations.get()
+            for i in range(number_accumulations):
+                self.ccd.atmcd64d.wait_for_acquisition()
+
+        # get and return spectrum
+        return self.ccd.atmcd64d.get_acquired_data(self.ccd.x_pixels)
+
+
+class Andor_DU401(Instrument):
+    """
+    Instrument driver for the Andor DU401 BU2 CCD.
+
+    Args:
+        name (str): Instrument name.
+        dll_path (str): Path to the atmcd64.dll file. If not set, a default path is used.
+        camera_id (int): ID for the desired CCD.
+        setup (bool): Flag for the setup of the CCD. If true, some default settings will be sent to the CCD.
```

Comment:
how "default" are these settings? In principle, the driver should not impose anything that is specific to only a subset of measurements that is supports. If it does, then in becomes much more difficult to implement other measurements (or measurement modes) without breaking user's code (who started relying on some defaults).

I'd strongly recommend to remove this "setup" feature and the corresponding flag, and let the users decide what the defaults are.